src/components/Main/Results/DeterministicLinePlot.tsx

```diff
@@ -248,103 +253,111 @@ export function DeterministicLinePlot({
+                    domain={[tMin, tMax]}
+                    tickCount={7}
+                  />
+
+                  <YAxis scale={logScaleString} type="number" domain={[1, 'dataMax']} tickFormatter={yTickFormatter} />
+
+                  <YAxis yAxisId="mitigationStrengthAxis" orientation={'right'} type="number" domain={[0, 1]} />
+
+                  <Tooltip
+                    formatter={tooltipFormatter}
+                    labelFormatter={labelFormatter}
+                    position={tooltipPosition}
+                    content={ResponsiveTooltipContent}
                   />
-                ))}
-
-                {scatterToPlot.map((d) => (
-                  <Scatter key={d.key} dataKey={d.key} fill={d.color} name={d.name} />
-                ))}
-              </ComposedChart>
-            </>
-          )
-        }}
-      </ReactResizeDetector>
-    </div>
-  )
+
+                  <Legend
+                    verticalAlign="top"
+                    formatter={(v, e) => legendFormatter(enabledPlots, v, e)}
+                    onClick={(e) => {
+                      const plots = enabledPlots.slice(0)
+                      enabledPlots.includes(e.dataKey)
+                        ? plots.splice(plots.indexOf(e.dataKey), 1)
+                        : plots.push(e.dataKey)
+                      setEnabledPlots(plots)
+                    }}
+                  />
+
+                  {mitigationIntervals.map((interval) => (
+                    <ReferenceArea
+                      key={interval.id}
+                      x1={_.clamp(interval.timeRange.tMin.getTime(), tMin, tMax)}
+                      x2={_.clamp(interval.timeRange.tMax.getTime(), tMin, tMax)}
+                      y1={0}
+                      y2={_.clamp(interval.mitigationValue, 0, 1)}
+                      yAxisId={'mitigationStrengthAxis'}
+                      fill={interval.color}
+                      fillOpacity={0.25}
+                    >
+                      <Label value={interval.name} position="insideTopRight" fill="#444444" />
+                    </ReferenceArea>
+                  ))}
+
+                  {linesToPlot.map((d) => (
+                    <Line
+                      key={d.key}
+                      dot={false}
+                      isAnimationActive={false}
+                      type="monotone"
+                      strokeWidth={3}
+                      dataKey={d.key}
+                      stroke={d.color}
+                      name={d.name}
+                      legendType={d.legendType}
+                    />
+                  ))}
+
+                  {scatterToPlot.map((d) => (
+                    <Scatter key={d.key} dataKey={d.key} fill={d.color} name={d.name} />
+                  ))}
+                </ComposedChart>
+              </>
+            )
+          }}
+        </ReactResizeDetector>
+      </div>
+    )
+  }
+
+  debouncedRender()
+
+  return rendered
```

Comment:
Invoked the debounced render to (maybe) update the render state. Then return the (maybe updated) render state.